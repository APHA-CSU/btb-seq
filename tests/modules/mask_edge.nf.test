nextflow_process {

    name "Test Process mask"
    script "main.nf"
    process "MASK"

    test("Should produce a combined mask in bed format") {

        when {
            params {
                rptmask = "${projectDir}/tests/data/edge-cases.bed"
                allsites = "${projectDir}/tests/data/edge-cases_regions.bed"
                MIN_READ_DEPTH = 8
                MIN_ALLELE_FREQUENCY_ALT = 0.8
                MIN_ALLELE_FREQUENCY_REF = 0.7
            }
            process {
                """
                input[0] = ["edge-cases", file("${projectDir}/tests/data/edge-cases.vcf.gz"), 
                            file("${projectDir}/tests/data/edge-cases.vcf.gz.csi"), 
                            file("${projectDir}/tests/data/edge-cases.sam")] 
                input[1] = params.rptmask
                input[2] = params.allsites
                input[3] = params.MIN_READ_DEPTH
                input[4] = params.MIN_ALLELE_FREQUENCY_ALT
                input[5] = params.MIN_ALLELE_FREQUENCY_REF
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}