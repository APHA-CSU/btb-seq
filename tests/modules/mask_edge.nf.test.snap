{
    "Should produce a combined mask in bed format for edge-cases": {
        "content": [
            {
                "0": [
                    [
                        "edge-cases",
                        "edge-cases.mask.bed:md5,5c0c4871b7491696a8b4bd101e2d0f3b",
                        "edge-cases.nonmasked-regions.bed:md5,ccec7730ac592e94730fd75f809aba2f"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-12T07:59:34.794506279"
    }
}